### ISSUE TITLE
Make request headers available on the error page

### ISSUE THREAD

**Issue by thijsw** [nuxt/framework#8083]
### Describe the feature

As described in this [discussion thread](https://github.com/nuxt/framework/discussions/6940), the request headers are not available on an error page. Therefore, cookies are not available either. So this code on a `~/error.vue` page yield both null values. I am using server-side rendering.

```ts
const host = useCookie('name')
const headers = useRequestHeaders(['user-agent', 'host'])
```

I can imagine that it would be really useful to have this information available on an error page.

A potential "fix" could be easily implemented by passing the request headers to the `$fetch` function on [this line](https://github.com/nuxt/framework/blob/main/packages/nuxt/src/core/runtime/nitro/error.ts#L47):

```ts
let html = !isErrorPage
  ? await $fetch(withQuery('/__nuxt_error', errorObject), {
      headers: event.req.headers,
    }).catch(() => null)
  : null
```

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by danielroe** [comment]
I completely understand the use case. The only question here is about caching, as if we pass headers, then we can't cache the error page result. (@pi0 - what do you think?)

**Comment by Luffyyy** [comment]
While I understand the less dependencies for error page does make sense especially if the error is a more "deep/core error", there are situations we just wanna do a 404 error and have everything under control. In my site at least, I do want to still show the user on the header and the headers being missing messes it up. Why not add an option for soft non-cachable errors?

**Comment by thijsw** [comment]
Fixed by #7340
